src/EnergyPlus/HeatBalanceManager.cc

```diff
@@ -4382,102 +4480,71 @@ namespace HeatBalanceManager {
-            }
-            thisConstruct.SourceSinkPresent = true;
-            thisConstruct.SourceAfterLayer = int(DummyProps(1));
-            thisConstruct.TempAfterLayer = int(DummyProps(2));
-            thisConstruct.SolutionDimensions = int(DummyProps(3));
-            if ((thisConstruct.SolutionDimensions < 1) ||
-                (thisConstruct.SolutionDimensions > 2)) {
-                ShowWarningError(state, "Construction:InternalSource must be either 1- or 2-D.  Reset to 1-D solution.");
-                ShowContinueError(state, "Construction=" + thisConstruct.Name + " is affected.");
-                thisConstruct.SolutionDimensions = 1;
-            }
-            thisConstruct.ThicknessPerpend = DummyProps(4) / 2.0;
-            thisConstruct.userTemperatureLocationPerpendicular = thisConstruct.setUserTemperatureLocationPerpendicular(state, DummyProps(5));
+        auto instances = inputProcessor->epJSON.find(CurrentModuleObject);
+        if (instances != inputProcessor->epJSON.end()) {
+            AnyInternalHeatSourceInInput = true;
+            auto &instancesValue = instances.value();
+            for (auto instance = instancesValue.begin(); instance != instancesValue.end(); ++instance) {
+                auto const &fields = instance.value();
+                auto const &thisObjectName = UtilityRoutines::MakeUPPERCase(instance.key());
 
-            // Set the total number of layers for the construction
-            thisConstruct.TotLayers = ConstructNumAlpha - 1;
-            if (thisConstruct.TotLayers <= 1) {
-                ShowSevereError(state, "Construction " + thisConstruct.Name +
-                                " has an internal source or sink and thus must have more than a single layer");
-                ErrorsFound = true;
-            }
-            if ((thisConstruct.SourceAfterLayer >= thisConstruct.TotLayers) ||
-                (thisConstruct.SourceAfterLayer <= 0)) {
-                ShowWarningError(state, "Construction " + thisConstruct.Name + " must have a source that is between two layers");
-                ShowContinueError(state, "The source after layer parameter has been set to one less than the number of layers.");
-                thisConstruct.SourceAfterLayer = thisConstruct.TotLayers - 1;
-            }
-            if ((thisConstruct.TempAfterLayer >= thisConstruct.TotLayers) ||
-                (thisConstruct.TempAfterLayer <= 0)) {
-                ShowWarningError(state, "Construction " + thisConstruct.Name +
-                                 " must have a temperature calculation that is between two layers");
-                ShowContinueError(state, "The temperature calculation after layer parameter has been set to one less than the number of layers.");
-                thisConstruct.TempAfterLayer = thisConstruct.TotLayers - 1;
-            }
+                std::string construction_name{UtilityRoutines::MakeUPPERCase(fields.at("construction_name"))};
```

Comment:
I think you need an `inputProcessor->markObjectAsUsed(CurrentModuleObject, instance.key());` to avoid the ERR diffs popping up on this branch.